### ISSUE TITLE
Squash/Rebase PR commits instead of merging

### ISSUE THREAD

**Issue by gekkowrld** [CsqLang/Csq#238]
Merging commits creates a really messy git tree that can be hard to follow.
I suggest using:

1. **rebase**: It keeps all the commits makes but in a nice linear timeline that is easy to follow.
2. **squash**: Commits only one commit from a pr, this reduces the number of commits that are sometimes make the main branch cluttered.

**Comment by AniketKumar2500** [comment]
But in squash, suppose you are creaing a new feature so if you are directly going to pull per commit that incomplete code might affect whole Csq's main branch code.

**Comment by gekkowrld** [comment]
In squashing you create one commit for the whole pr. So in context of creating a feature, after you create a feature in a pr and you are satisfied that is ready to be merged to the main branch, you squash all the commits to one commit and then merge it.
